### PR TITLE
Bundle App for Desktop and Upload in CI

### PR DESCRIPTION
**Pull request by nickfarrow**
App build matrix:
- Build AppImage for linux
    - runs on arch linux 
    - runs on ubuntu-18 in a docker container (no graphical display but looked good)
- APK and app bundle (needed for android stores)

These artifacts get uploaded during the action.

Builds upon https://github.com/frostsnap/frostsnap/pull/165

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -78,9 +113,12 @@ jobs:
...
+      - run: sudo apt-get update && sudo apt install -y libudev-dev
       - run: just frostsnapp/install-rust-deps
-      - run: just frostsnapp/build linux
-      - run: just frostsnapp/build appbundle
+      - run: just frostsnapp/build appbundle --release
+      - name: Upload APK
+        uses: actions/upload-artifact@v4
+        with:
+          name: frostsnapp-${{ github.sha }}.apk
+          path: target/riscv32imc-unknown-none-elf/release/v2
```
Comment (LLFourn) [inline .github/workflows/test.yml:123]: This is called `Upload APK` but it uploads an `riscv32imc` elf binary?

**Comment by LLFourn** [comment]
@LLFourn to check out whether MacOS works.

**Comment by LLFourn** [comment]
Let's be slightly less ambitious and just publish android and linux for now. We can try to iron out windows, mac and iPhone later.

**Comment by nickfarrow** [comment]
TODOs:
- [ ] fix appimage size
- [ ] check without appimage
- [ ] check if we need canberra-gtk

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -375,11 +373,12 @@ jobs:
         with:
           path: ~/.pub-cache/
           key: ${{ runner.os }}-pub-cache-${{ hashFiles('frostsnapp/pubspec.lock') }}-v1
+          restore-keys: |
+            ${{ runner.os }}-pub-cache-
 
       - uses: subosito/flutter-action@v2
         with:
           channel: stable
-          cache: true
```
Comment (nickfarrow) [inline .github/workflows/test.yml:318]: i think this looks for **/pubspec.lock, which doesn't work? Will revisit tomorrow
https://github.com/frostsnap/frostsnap/actions/runs/11666583441/job/32481995066

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -259,35 +279,55 @@ jobs:
...
+            libcairo2-dev \
+            python3-pip \
+            git \
+            wget \
+            libudev-dev \
+            qtbase5-dev \
+            qt5-qmake \
+            qtdeclarative5-dev \
+            cmake \
+            libarchive-tools
```
Comment (LLFourn) [inline .github/workflows/test.yml:329]: Why do we have all these dependencies now? Also wget and get? I think they are already installed. This seems to be something to do with app image stuff? (Although I don't really understand why even app images would need all these qt and libcairo etc). If so it should be done in another step. The app image just needs the flutter linux build artifact right? We should build that in one step and package the app image in a different workflow IMO.


Also can we have a `just` command to build the app image and have the CI call the just command.

File: .github/workflows/test.yml (hunk 2 above)
Comment (LLFourn) [reply .github/workflows/test.yml:318]: we can just commit `pubspec.lock`. It would be better to use the flutter action caching approach.

Comment (nickfarrow) [reply .github/workflows/test.yml]: Was already commited. I just undid this change and it magically worked. I also deleted all the caches, ran again, and it still worked.

**Review by LLFourn — APPROVED** [review]
self-ACK. @nickfarrow do the honors of merging if you're good with this.

**Pull request merged**